tetris.py

```diff
@@ -292,10 +316,132 @@ def full_line_clear(self):
 
 
+class Board:
+    def __init__(self, canvas, outline_color, width, height, block_shape, block_center, rotate_counter):
+        self.canvas = canvas
+        self.outline_color = outline_color
+        self.width = width
+        self.height = height
+        self.block_shape = block_shape
+        self.block_center = block_center
+        self.rotate_counter = rotate_counter
+
+    def draw_board(self):
+        """
+        Draws the board of rectangles on top of the canvas
+        """
+        x_gap = 0
+        for x in range(self.width):
+            y_gap = 0
+            for y in range(self.height):
+                self.canvas.create_rectangle(
+                    x_gap,
+                    y_gap,
+                    x_gap + square_size,
+                    y_gap + square_size,
+                    fill=D_GREY,
+                    outline=self.outline_color,
+                )
+                y_gap += square_size
+
+            x_gap += square_size
+
+    def draw_block(self, landed_blocks):
+        """
+        Draws the different shapes on the board
+        """
+        self.color_dict = {
+            "L": YELLOW,
+            "I": RED,
+            "E": GREEN,
+            "L_rev": BLUE,
+            "Z": PURPLE,
+            "Z_rev": TEAL,
+            "O": ORANGE,
+        }
+        self.canvas.delete("block")
+        self.canvas_small.delete('block')
+
+        for x, y in self.get_block_shape(self.tetris_game.current_block_shape, self.tetris_game.current_block_center, self.tetris_game.rotate_counter):
+            self.canvas.create_rectangle(
+                x * square_size,
+                y * square_size,
+                x * square_size + square_size,
+                y * square_size + square_size,
+                tags="block",
+                fill=self.color_dict[self.tetris_game.current_block_shape],
+            )
+
+        for letter, coord_list in self.tetris_game.landed_blocks.items():
+            for (x, y) in coord_list:
+                self.canvas.create_rectangle(
+                    x * square_size,
+                    y * square_size,
+                    x * square_size + square_size,
+                    y * square_size + square_size,
+                    tags="block",
+                    fill=self.color_dict[letter],
+                )
+
+        for x, y in self.get_block_shape(self.tetris_game.upcoming_block_shape, self.tetris_game.upcoming_block_center, 0):
+            self.canvas_small.create_rectangle(
+                    x * square_size,
+                    y * square_size,
+                    x * square_size + square_size,
+                    y * square_size + square_size,
+                    tags="block",
+                    fill=self.color_dict[self.tetris_game.upcoming_block_shape],
+            )
+
+    def get_block_shape(self, block_shape, block_center, rotate_counter):
+        (x, y) = block_center
```

Comment:
And instead use stuff conveniently from `self`:

```suggestion
        (x, y) = self.block_center
```